rust/rsmgp-sys/src/property/mod.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2016-2021 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ffi::CString;
+
+use crate::memgraph::*;
+use crate::mgp::*;
+use crate::value::*;
+// Required here, if not present tests linking fails.
+#[double]
+use crate::mgp::ffi;
+use mockall_double::double;
+
+/// All related to the property graph property (data key-value pair).
+///
+/// Property is used in the following contexts:
+///   * return Property from PropertiesIterator
+///   * return Property directly from vertex/edge.
+///
+/// Property owns CString and Value bacause the underlying C string or value could be deleted
+/// during the lifetime of the property. In other words, Property stores copies of underlying name
+/// and value.
+#[derive(Debug)]
+pub struct Property {
+    pub name: CString,
```

Comment:
If we decide to go with the "read without copy" approach, then this should be replaced by `CStr`.

Reply:
The same goes for `MapItem::key` .

Reply:
We've abandoned the idea with 0 copies because of an issue with the lifetime of C and Rust objects. Take a look into the `mgp_raw_value_to_value` function comment.